scriptmodules/supplementary/launchingimages.sh

```diff
@@ -0,0 +1,413 @@
+          1 "Use the system color defined by theme"
+          2 "Select a color"
+    )
+    choice=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+
+    case "$choice" in
+        0)  echo
+            ;;
+        1)  echo "--solid-bg-color"
+            ;;
+        2)  echo "--solid-bg-color $(_select_color_launchingimages background)"
+            ;;
+        *)  echo "\$solid_bg_color"
+            ;;
+    esac
+}
+
+function _dialog_yesno_launchingimages() {
+    dialog --backtitle "$__backtitle" --yesno "$@" 20 60 2>&1 >/dev/tty
+}
+
+function _set_no_ask_launchingimages() {
+    _dialog_yesno_launchingimages "If you enable \"no_ask\" all generated images will be automatically accepted.\n\nDo you want to enable it?" \
+    && echo "--no-ask"
+}
+
+function _set_no_logo_launchingimages() {
+    _dialog_yesno_launchingimages "If you enable \"no_logo\" the images won't have the system logo (useful for tronkyfran theme, for example).\n\nDo you want to enable it?" \
+    && echo "--no-logo"
+}
+
+function _set_logo_belt_launchingimages() {
+    _dialog_yesno_launchingimages "If you enable \"logo_belt\" the image will have a semi-transparent white belt behind the logo.\n\nDo you want to enable it?" \
+    && echo "--logo-belt"
+}
+
+function _get_all_launchingimages() {
+    find "$configdir" -type f -regex ".*launching\.\(png\|jpg\)" | sort
+}
+
+function _is_theme_chosen_launchingimages() {
+    if [[ -z "$1" ]]; then
+        printMsgs "dialog" "You didn't choose a theme!\n\nGo to the \"Image generation settings\" and choose one."
+        return 1
+    fi
+}
+
+function _get_config_file_launchingimages() {
+    local file_list=$(find "$md_inst" -type f -name '*.cfg' ! -name '.current.cfg' | sort | xargs)
+    if [[ -z "$file_list" ]]; then
+        printMsgs "dialog" "There's no config file saved."
+        return 1
+    fi
+    _dialog_menu_launchingimages "Choose the file" $file_list # XXX: no quotes surrounding $file_list is mandatory!
+    return $?
+}
+
+function _load_config_launchingimages() {
+    echo "$(loadModuleConfig \
+        'theme=' \
+        'extension=png' \
+        'show_timeout=5' \
+        'loading_text=NOW LOADING' \
+        'press_button_text=PRESS A BUTTON TO CONFIGURE LAUNCH OPTIONS' \
+        'loading_text_color=white' \
+        'press_button_text_color=gray50' \
+        'no_ask=' \
+        'no_logo=' \
+        'solid_bg_color=' \
+        'system=' \
+        'logo_belt=' \
+    )"
+}
+
+function _settings_launchingimages() {
+    local cmd=(dialog --backtitle "$__backtitle" --title " SETTINGS " --menu "runcommand launching images generation settings." 22 86 16)
+    local options
+    local choice
+    local config_file
+
+    iniConfig ' = ' '"' "$md_inst/.current.cfg"
+
+    while true; do
+        eval $(_load_config_launchingimages)
+
+        options=( 
+            config_file "$(
+               [[ "$config_file" =~ $theme\.cfg$ ]] && echo "$(basename "$config_file")"
```

Comment:
can you avoid a re here ? eg something like

`[[ "$config_file" == *"$theme.cfg" ]] && echo "$(basename "$config_file")"`